File: day2/src/Part1Test.java
import static org.junit.jupiter.api.Assertions.*;

class Part1Test {

    @org.junit.jupiter.api.Test
    void calculateLineChecksum() {
        // Arrange
        Part1 p1 = new Part1();

        // Act
        int checksum1 = p1.calculateLineChecksum("5\t1\t9\t5");
        int checksum2 = p1.calculateLineChecksum("7\t5\t3");
        int checksum3 = p1.calculateLineChecksum("2\t4\t6\t8");

        // Assert
        assertEquals(8, checksum1);
        assertEquals(4, checksum2);
        assertEquals(6, checksum3);
    }
}

File: day3/src/Part1Test.kt
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*

internal class Part1Test {

    @Test
    fun calculate() {
        // Arrange
        val part1 = Part1()

        // Act
        val test1 = part1.calculate(1)
        val test2 = part1.calculate(12)
        val test3 = part1.calculate(23)
        val test4 = part1.calculate(1024)

        // Assert
        kotlin.test.assertEquals(0, test1)
        kotlin.test.assertEquals(3, test2)
        kotlin.test.assertEquals(2, test3)
        kotlin.test.assertEquals(31, test4)
    }
}
